src/include/mpir_datatype.h

```diff
@@ -329,7 +329,7 @@ void MPIR_Datatype_get_flattened(MPI_Datatype type, void **flattened, int *flatt
 #define MPIR_Datatype_add_ref_if_not_builtin(datatype_)             \
```

Comment:
I opted not to change the name of the macro in this commit. It is used extensively in the code, so would be a lot of churn, and it's likely the intent was to exempt pairtypes all along.